service/frontend/validators.go

```diff
@@ -39,3 +41,47 @@ func validateStringField(fieldName string, value string, maxLen int, required bo
 	}
 	return nil
 }
+
+func validateFairnessWeightUpdate(
+	set map[string]float32,
+	unset []string,
+	maxConfigLimit int,
+) error {
+	total := len(set) + len(unset)
+	if total > maxConfigLimit {
+		return serviceerror.NewInvalidArgumentf(
+			"too many fairness weight overrides in request: got %d, maximum %d",
+			total, maxConfigLimit,
+		)
+	}
+
+	for k, w := range set {
+		if strings.TrimSpace(k) == "" {
+			return serviceerror.NewInvalidArgument(
+				"fairness weight override key must not be empty")
+		}
+		if w <= 0 {
+			return serviceerror.NewInvalidArgumentf(
+				"invalid fairness weight weight for key %q: must be greater than zero", k,
+			)
+		}
+	}
```

Comment:
We should use the existing code in `common/priorities` to validate (and refactor it if necessary). This is missing length validation.